provider/lxd/environ_raw.go

```diff
@@ -15,30 +16,37 @@ import (
 	jujulxdclient "github.com/juju/juju/tools/lxdclient"
 )
 
+// TODO (manadart 2018-05-09) This is really nothing but an LXD client and does
+// not need its own type.
+// As the LXD facility is refactored, this will be removed altogether.
+// As an interim measure, the new and old client implementations will be have
+// interface shims.
+// After the old client is removed, provider tests can be rewritten using
+// GoMock, at which point rawProvider is replaced with the new client.
 type rawProvider struct {
+	newClient
 	lxdCerts
-	lxdConfig
 	lxdInstances
 	lxdProfiles
-	lxdImages
 	lxdStorage
 
 	remote jujulxdclient.Remote
 }
 
+type newClient interface {
+	FindImage(string, string, []lxd.RemoteServer, bool, environs.StatusCallbackFunc) (lxd.SourcedImage, error)
+	GetServer() (server *lxdapi.Server, ETag string, err error)
+	GetConnectionInfo() (info *lxdclient.ConnectionInfo, err error)
+	UpdateServerConfig(map[string]interface{}) error
```

Comment:
The SetServerConfig that is being replaced took both keys and values as strings. Can we be consistent?

Reply:
See previous comment based on different config types. Should be safe to use strings; I'll change it.